File: ml_service/scratch/list_models.py
import os
from google import genai
from dotenv import load_dotenv

load_dotenv(dotenv_path=os.path.join(os.getcwd(), '..', 'server', '.env'))
GEMINI_API_KEY = os.getenv("GEMINI_API_KEY")

client = genai.Client(api_key=GEMINI_API_KEY)

try:
    for model in client.models.list():
        print(f"Model: {model.name}, Supported: {model.supported_actions}")
except Exception as e:
    print("Error listing models:", e)

File: server/scratch/list-models.js
import fetch from 'node-fetch';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
dotenv.config({ path: path.join(__dirname, '../.env') });

const apiKey = process.env.GEMINI_API_KEY || process.env.AI_API_KEY;

async function listModels(apiVersion) {
    const url = `https://generativelanguage.googleapis.com/${apiVersion}/models?key=${apiKey}`;
    try {
        const res = await fetch(url);
        const status = res.status;
        const json = await res.json();
        console.log(`[${apiVersion}] ListModels status: ${status}`);
        if (status === 200 && json.models) {
            console.log('Available models:');
            json.models.forEach(m => {
                console.log(`  - Name: ${m.name}`);
                console.log(`    Supported methods: ${m.supportedGenerationMethods.join(', ')}`);
            });
        } else {
            console.log(`  Failed:`, JSON.stringify(json).substring(0, 300));
        }
    } catch (err) {
        console.error(`  Error: ${err.message}`);
    }
}

async function run() {
    await listModels('v1beta');
    console.log('----------------------------');
    await listModels('v1');
}

run();
